### PR TITLE
Fix ResetGlobalProperties help prompt parameter

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixed `ResetGlobalProperties` help prompt request parameter reset to default value. 
The root cause of this defect is that SDL does not get default help prompt value from Application Manager settings and does not assign it to app's help prompt. It just assigns an empty map.

In this pull request:
- Added missed `help_prompt()` function to `ApplicationManagerSettings` interface
- Added related mock method to `MockApplicationManagerSettings` class
- Updated `ResetGlobalPropertiesRequest::ResetHelpPromt()` function logic to work as expected
- Updated related unit tests

### CODE REVIEW

**Comment by AKalinich-Luxoft** [comment]
@AByzhynar, @anosach-luxoft, @VProdanov, @dev-gh please review

**Comment by dev-gh** [comment]
@AKalinich-Luxoft 
Put issue link in description

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc [hunk 1]
```diff
@@ -194,6 +194,19 @@ bool ResetGlobalPropertiesRequest::ResetHelpPromt(
   }
   smart_objects::SmartObject so_help_prompt =
       smart_objects::SmartObject(smart_objects::SmartType_Array);
+
+  const std::vector<std::string>& help_prompt_default =
+      application_manager_.get_settings().help_prompt();
+
+  for (uint32_t i = 0; i < help_prompt_default.size(); ++i) {
```
Comment (AByzhynar) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @AKalinich-Luxoft Don't compare different types even both of them unsigned. Type sizes can be different.

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 2]
```diff
@@ -149,12 +149,23 @@ TEST_F(ResetGlobalPropertiesRequestTest, Run_InvalidVrHelp_UNSUCCESS) {
       mobile_apis::GlobalProperty::KEYBOARDPROPERTIES;
 
   EXPECT_CALL(app_mngr_, RemoveAppFromTTSGlobalPropertiesList(kConnectionKey));
-  smart_objects::SmartObject so_prompt =
+
+  std::vector<std::string> help_prompt{"help_text"};
```
Comment (dev-gh) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @AKalinich-Luxoft 
This is C++11 feature, is that standard already allowed for open SDL?

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc (hunk 1 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @AByzhynar, replaced with iterators. See https://github.com/smartdevicelink/sdl_core/pull/1555/commits/a5b4a71170dd4266d19774aecc787cbc91f8bb08

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 2 above)
Comment (AKalinich-Luxoft) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dev-gh oficially not. I've removed brace initialization in https://github.com/smartdevicelink/sdl_core/pull/1555/commits/a5b4a71170dd4266d19774aecc787cbc91f8bb08

**Comment by AKalinich-Luxoft** [comment]
@dev-gh AFAIK we should not attach jira-links to pull requests in this project

**Comment by AByzhynar** [comment]
@AKalinich-Luxoft Don't forget to squash commits

**Comment by AByzhynar** [comment]
@AKalinich-Luxoft Why you did not add somebody from OpenSDL team to code review?

**Comment by AKalinich-Luxoft** [comment]
@AByzhynar I add only reviewers from updated table:
https://adc.luxoft.com/confluence/pages/viewpage.action?pageId=309925667

**Comment by dboltovskyi** [comment]
According to req-t "[ResetGlobalProperties] SUCCESS on UI.SetGlobalProperties and TTS.SetGlobalPrtoperties":
>In case mobile app sends valid ResetGlobalProperties with "HELPPROMPT" and/or "TIMEOUTPROMPT" and at least one other valid parameter in "properties" array, SDL must:
>1) set-up the value of 'HelpPrompt' to an empty array and/or retrieve the value of "TimeOutPrompt" parameter from .ini file correspondingly
>2) transfer the items requested ("HELPPROMPT" or/and "TIMEOUTPROMPT") TTS.SetGlobalProperties(helpPrompt:"<empty array>", and/or timeoutPrompt:<'TimeOutPrompt' from ini.file>) to HMI.
>3) transfer UI.SetGlobalProperties with appropriate values to HMI (see more APPLINK-20718, APPLINK-20825).
>4) respond (resultCode:SUCCESS, success:true) to mobile application on getting the both SUCCESS:UI.SetGlobalProperties and SUCCESS:TTS.SetGlobalProperties from HMI

Value of "helpPrompt" has to be empty
So related issue is not actual

**Pull request closed without merging**